src/cdk/accordion/accordion-item.ts

```diff
@@ -30,6 +31,11 @@ let nextId = 0;
 @Directive({
   selector: 'cdk-accordion-item, [cdkAccordionItem]',
   exportAs: 'cdkAccordionItem',
+  providers: [
+    // Provide CdkAccordion as undefined to prevent nested accordion items from registering
+    // to the same accordion.
+    {provide: CdkAccordion, useValue: undefined},
```

Comment:
TIL that you can do this. I had to do something similar on the menu, but I ended up adding a provider and having the items register themselves with it.